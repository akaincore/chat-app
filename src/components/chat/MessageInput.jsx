import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  container: {
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    bottom: '10px',
    width: 'calc(100% - 320px)',
  },
  input: {
    margin: '20px 10px',
    width: '98%',
  },
  paper: {
    width: '90%',
  }
});

class MessageInput extends React.Component {

  state = {
    message: '',
  };

  onMessageChange = (event) => {
    this.setState({
      message: event.target.value,
    });
  };

  onKeyPress = (event) => {
    const { message } = this.state;

    if (event.key === 'Enter' && message) {
      this.props.sendMessage(message);
      this.setState({ message: '' });
    }
  };

  render() {
    const {
      classes,
      joinChat,
      currentUser,
      disabled,
    } = this.props;
    const { message } = this.state;
    return (
      <div className={classes.container}>
        <Paper className={classes.paper}>
          {currentUser.isChatMember &&
          <Input
            disabled={disabled}
            fullWidth={true}
            placeholder="Type message..."
            className={classes.input}
            inputProps={{
              'aria-label': 'Description',
            }}
            value={message}
            onChange={this.onMessageChange}
            onKeyPress={this.onKeyPress}
          />
          }
          {!currentUser.isChatMember &&
          <Button
            disabled={disabled}
            onClick={joinChat}
            fullWidth
            variant="contained"
            color="primary"
          >
            Join
          </Button>
          }
        </Paper>
      </div>
    );
  };

}

export default withStyles(styles)(MessageInput);
