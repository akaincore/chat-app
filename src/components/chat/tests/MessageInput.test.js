/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from '../MessageInput';

const mockProps = {
  joinChat: jest.fn(),
  sendMessage: jest.fn(),
  currentUser: {
    username: 'user',
    isMember: false,
    isChatMember: false,
    isCreator: false,
  },
  disabled: false,
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
