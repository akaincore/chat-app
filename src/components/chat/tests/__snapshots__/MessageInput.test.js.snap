// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders correctly 1`] = `
<div
  className="MessageInput-container-67"
>
  <div
    className="MuiPaper-root-70 MuiPaper-elevation2-74 MuiPaper-rounded-71 MessageInput-paper-69"
  >
    <button
      className="MuiButtonBase-root-123 MuiButton-root-97 MuiButton-contained-108 MuiButton-containedPrimary-109 MuiButton-raised-111 MuiButton-raisedPrimary-112 MuiButton-fullWidth-122"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-98"
      >
        Join
      </span>
      <span
        className="MuiTouchRipple-root-126"
      />
    </button>
  </div>
</div>
`;
